sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -100,6 +70,23 @@ package object util extends Logging {
     file
   }
 
+  private def toByteArray(inStream: InputStream): Array[Byte] = {
+    val outStream = new ByteArrayOutputStream
+    try {
+      var reading = true
+      while (reading) {
+        inStream.read() match {
```

Comment:
OK, hah, now that I look at it - this is a pretty inefficient way to copy the byte stream. It should be read and written in chunks, not a byte at a time.

That's not hard to implement, but Guava already does this:
https://github.com/google/guava/blob/a0e2577de61a0d7e8a3dd075be66a31c93ea0446/android/guava/src/com/google/common/io/ByteStreams.java#L173

In fact we use ByteStreams.toByteArray in several places. Just use that? it'll be simpler and more efficient

Reply:
It has been changed to call `ByteStreams.toByteArray`, but do we still need to extract a private method? It seems that there are not many duplicate codes

Reply:
Yes, it looks better